devolutions-crypto-wayk/src/bastion/key_derive.rs

```diff
@@ -0,0 +1,94 @@
+use hkdf::Hkdf;
+use sha2::Sha256;
+use uuid::Uuid;
+
+/// Derives tenant key from subscription master key and tenant UUID
+pub fn derive_tenant_key(
+    subscription_id: Uuid,
+    tenant_id: Uuid,
+    subscription_master_key: &[u8],
+) -> Vec<u8> {
+    derive_key_impl(subscription_id, tenant_id, subscription_master_key, 64)
```

Comment:
Since the tenant separation is binded to the "info" parameters, I'm going to verify in the RFC that there's a guarantee and that it is not forgeable. Didn't think about it while reviewing the paper. Eyeballing for now, I think inverting Info and Salt would be better, but I'll go deeper into it.

Reply:
Let us know!

Reply:
Info is good for this use case! I'm resolving this issue.